### PR TITLE
Backup mongo

### PR DESCRIPTION
**Pull request by jbygdell**
* backup and restore of mongodb
* local test script
* option to run test script with a container
* container specific GH action
* updated readme

related to NBISweden/LocalEGA-SE-Deployment#166

### CODE REVIEW

File: dev_tools/scripts/mongo_test.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+num=100
+
+if [[ -n $1 ]]; then
```
Comment (pontus) [inline dev_tools/scripts/mongo_test.sh:5]: This obviously works, but I'm always more comfortable with protecting variables with quotes.

Comment (jbygdell) [reply dev_tools/scripts/mongo_test.sh:5]: Currently bashing my head against testing with TLS enabled.

File: mongo.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+		log.Error("Could not initialise decryptor", err)
+		return err
+	}
+	d, err := newDecompressor(key, r)
+	if err != nil {
+		log.Error("Could not initialise decompressor", err)
+		return err
+
+	}
+	data, err := ioutil.ReadAll(d)
```
Comment (pontus) [inline mongo.go:102]: `ReadFrom` into `in` instead? (Avoids having an extra copy needing to be present in memory, or one can go for a pipe to save even more :).

**Review by pontus — DISMISSED** [review]
:+1:

**Comment by pontus** [comment]
I'm not sure I understand the difference uses for `mongo_test.sh` and `mongo_test_docker.sh`, can they be united somehow?

**Comment by jbygdell** [comment]
> I'm not sure I understand the difference uses for `mongo_test.sh` and `mongo_test_docker.sh`, can they be united somehow?

When running the test from within the docker network we can test replica set connectivity by requesting `readPreference=secondary`, for that to work we need to be able to support dns redirects

**Comment by pontus** [comment]
Now I cloned and made a diff, feels like that could be handled by passing an option for whatever to run in docker or not. Alternatively moving common stuff to a separate script.

**Comment by jonandernovella** [comment]
What about just installing the mongo client on the test machine? Then there is no need to do all the `docker exec` magic.

**Pull request merged**